### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by willwhsiao**
This MR implements docSearch on the homepage of this doc site.

The API key used in the repo is a read only key and is not secret. Right now it is linked up to docusaurus' v2 docs site index for testing purposes.

TODO:
1. Once official docs are published at docs.openraven.com run the algolia docscraper with a write API key from our algolia prod application.
2. Switch the read API key and indexName in docusaurus.config.js

Do not merge until these blockers are complete.

Screenshots:
Screen Shot 2020-01-31 at 10 22 28 AM

### CODE REVIEW

File: openraven-theme/src/theme/HomePageSearchBar/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React, {useState, useRef } from 'react';
```
Comment (gabewomble) [inline openraven-theme/src/theme/HomePageSearchBar/index.js]: bracket spacing

File: openraven-theme/src/theme/HomePageSearchBar/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import classnames from 'classnames';
+
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import {useHistory} from '@docusaurus/router';
```
Comment (gabewomble) [inline openraven-theme/src/theme/HomePageSearchBar/index.js]: I'll just leave this as a general comment for anything this applies to -- are we not using an eslint config for this repo? If so, can we make it more consistent w/ the ones we use on other repos (doesn't have to be this PR)

File: openraven-theme/src/theme/HomePageSearchBar/algolia.css [hunk 3]
```diff
@@ -0,0 +1,529 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* Bottom border of each suggestion */
+.algolia-docsearch-suggestion {
+  border-bottom-color: #3a3dd1;
```
Comment (gabewomble) [inline openraven-theme/src/theme/HomePageSearchBar/algolia.css]: Would it make sense to set vars for these colors at more global level?

File: openraven-theme/src/theme/HomePageSearchBar/algolia.css [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+}
+/* Main category headers */
```
Comment (gabewomble) [inline openraven-theme/src/theme/HomePageSearchBar/algolia.css]: Imo lots of stuff in this file could be set as css vars for cleaner re-use

Comment (willwhsiao) [reply openraven-theme/src/theme/HomePageSearchBar/algolia.css]: This is mostly boiler plate from the official docusaurus repo. I just had to change a couple of lines here and there.

File: openraven-theme/src/theme/HomePageSearchBar/index.js (hunk 2 above)
Comment (willwhsiao) [reply openraven-theme/src/theme/HomePageSearchBar/index.js]: A lot of the code in here is written by docusaurus. I can see if they have a linter config they use.

**Pull request merged**